src/helpers/setUpEdgeFunction.js

```diff
@@ -0,0 +1,89 @@
+const { Buffer } = require('node:buffer')
+const { existsSync, readdirSync } = require('node:fs')
+const { writeFile, mkdir, readFile } = require('node:fs/promises')
+const { join, relative, sep, posix } = require('node:path')
+
+const { readJson } = require('fs-extra')
+
+const packageJson = require('../../package.json')
+
+/**
+ * Recursively lists all files in a directory.
+ */
+const getAllFilesIn = (dir) =>
+  readdirSync(dir, { withFileTypes: true }).flatMap((dirent) => {
+    if (dirent.isDirectory()) {
+      return getAllFilesIn(join(dir, dirent.name)).map((path) => join(dirent.name, path))
+    }
+    return [join(dir, dirent.name)]
+  })
+
+const toPosix = (path) => path.split(sep).join(posix.sep)
+
+const setUpEdgeFunction = async ({ angularJson, projectName, netlifyConfig, constants, failBuild }) => {
+  const project = angularJson.projects[projectName]
+  const {
+    architect: { build },
+  } = project
+  const outputDir = build?.options?.outputPath
+  if (!outputDir || !existsSync(outputDir)) {
+    return failBuild('Could not find build output directory')
+  }
+
+  console.log(`Writing Angular SSR Edge Function for project "${projectName}" ...`)
+
+  netlifyConfig.build.publish = join(outputDir, 'browser')
+
+  const edgeFunctionDir = join(constants.INTERNAL_EDGE_FUNCTIONS_SRC, 'angular-ssr')
+  await mkdir(edgeFunctionDir, { recursive: true })
+
+  const serverDistRoot = join(outputDir, 'server')
+  const html = await readFile(join(serverDistRoot, 'index.server.html'), 'utf-8')
+  const staticFiles = getAllFilesIn(join(outputDir, 'browser')).map(
+    (path) => `/${relative(join(outputDir, 'browser'), path)}`,
+  )
+
+  const { routes: prerenderedRoutes } = await readJson(join(outputDir, 'prerendered-routes.json'))
+  const excludedPaths = [...staticFiles, ...prerenderedRoutes]
+
+  // buy putting this into a separate module that's imported first,
+  // we ensure this is initialised before any other module
+  const polyfills = `
+  import process from "node:process"
+
+  globalThis.process = process
+  globalThis.global = globalThis
+  `
+
+  const ssrFunction = `
+  import "./polyfill.mjs";
+  import { Buffer } from "node:buffer";
+  import { renderApplication } from "${toPosix(relative(edgeFunctionDir, serverDistRoot))}/render-utils.server.mjs";
```

Comment:
I think you can do `path.posix.relative()` rather than implementing it yourself

Reply:
I can, but `edgeFunctionDir` and `serverDistRoot` will contain windows-style separators on windows. Not sure if `path.posix.relative()` know how to deal with them

Reply:
Ah, good point

Reply:
<img width="305" alt="Screenshot 2023-11-07 at 14 41 47" src="https://github.com/netlify/netlify-plugin-angular-universal/assets/14912729/df214913-a155-411c-a80a-2b6192d24900">

don't think that will work